### PR TITLE
Fix build 32-bit for Linux

### PR DESCRIPTION
**Pull request by rsetiawan7**
I've added a flag to build 32-bit .so file for Linux. The build was success, but not tested yet if something bad happen. And add missing includes too (since rewrite to C++17 standard).

### CODE REVIEW

**Comment by rsetiawan7** [comment]
I've updated the readme about build from source.

File: makefile [hunk 1]
```diff
@@ -15,7 +15,7 @@ hash:
 	$(GPP) -m32 -O2 -fshort-wchar -shared -o $(OUTPUT) *.o $(LIBRARIES)
 	
 cryptolib:
-	$(MAKE) -C $(CRYPTOPP_SRC_DIR) static
+	CXXFLAGS="-DNDEBUG -m32 -g2 -O3 -fPIC" $(MAKE) -C $(CRYPTOPP_SRC_DIR) static
```
Comment (rsetiawan7) [inline makefile]: It builds crypto++ in 32-bit. That's why the compiler says incompatible with crypto++ after build.

**Comment by Mellnik** [comment]
Tested on Ubuntu 18.04 and it works. Thanks for your PR.

**Comment by rsetiawan7** [comment]
You're welcome.

**Pull request merged**